src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.CodeFixes/IdentifierUpgradeCodeFixer.cs

```diff
@@ -22,32 +21,44 @@ public abstract class IdentifierUpgradeCodeFixer : CodeFixProvider
 
         public sealed override FixAllProvider GetFixAllProvider()
         {
+            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
             return WellKnownFixAllProviders.BatchFixer;
         }
 
         public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
+            // Get the syntax root of the document to be fixed
             var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
 
             if (root is null)
             {
                 return;
             }
 
-            var node = root.FindNode(context.Span);
+            // The node is found by location, so it's possible `FindNode` will find a parent with the same span.
+            // To get the correct node, find the first name or MAE in the given span.
+            var node = root.FindNode(context.Span)
+                .DescendantNodesAndSelf(n => true)
+                .FirstOrDefault(n => n.Span.Equals(context.Span)
+                    && (n is CSSyntax.NameSyntax
```

Comment:
seems like a candidate for language agnostic helper functions

Reply:
Yea, I can see that. I'll add extension methods.